File: src-tauri/src/commands/external/bttv.rs
use crate::dto::external::emote::EmoteEntry;
use crate::services;
use serde::Deserialize;

#[tauri::command]
pub async fn bttv_get_global_emotes() -> Result<Vec<EmoteEntry>, String> {
    services::external::bttv::get_global_emotes().await
}

#[derive(Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct BttvGetChannelEmotesParams {
    pub channel_id: String,
}

#[tauri::command]
pub async fn bttv_get_channel_emotes(
    params: BttvGetChannelEmotesParams,
) -> Result<Vec<EmoteEntry>, String> {
    services::external::bttv::get_channel_emotes(params.channel_id).await
}

File: src/lib/api/external/bttv.ts
import type { EmoteEntry } from "../../types/external/emote.ts";
import { invokeCommand, type InvokeOptions } from "../utils.ts";

export function bttvGetGlobalEmotes(options?: InvokeOptions): Promise<EmoteEntry[]> {
  return invokeCommand("bttv_get_global_emotes", undefined, options);
}

export type BttvGetChannelEmotesParams = {
  channelId: string;
};

export function bttvGetChannelEmotes(
  params: BttvGetChannelEmotesParams,
  options?: InvokeOptions,
): Promise<EmoteEntry[]> {
  return invokeCommand("bttv_get_channel_emotes", params, options);
}
